### PR TITLE
tests: Use vk_testing:: objects in descriptor tests

### PR DESCRIPTION
**Pull request by jeremyg-lunarg**
Add some stuff to make the vk_testing:: objects easier to use:

- vk_testing::*::destroy() methods 
- vk_testing::Buffer::address() and another constructor that makes it easy to set VK_MEMORY_ALLOCATE_DEVICE_ADDRESS_BIT

Then convert a bunch of descriptor tests to use these objects instead of raw vulkan handles.

### CODE REVIEW

**Review by spencer-lunarg — APPROVED** [review]
LGTM, really reduces a lot of code in a nice way!

File: tests/vktestbinding.cpp [hunk 1]
```diff
@@ -46,12 +47,14 @@ namespace {
     } while (0)
 
 #define NON_DISPATCHABLE_HANDLE_DTOR(cls, destroy_func) \
-    cls::~cls() noexcept {                              \
-        if (initialized()) {                            \
-            destroy_func(device(), handle(), NULL);     \
-            handle_ = VK_NULL_HANDLE;                   \
+    void cls::destroy() {                               \
+        if (!initialized()) {                           \
```
Comment (juan-lunarg) [inline tests/vktestbinding.cpp:51]: Shouldn't `destroy` also be marked an `noexcept`?

File: tests/vktestbinding.h [hunk 2]
```diff
@@ -519,6 +532,8 @@ class Buffer : public internal::NonDispHandle<VkBuffer> {
         return barrier;
     }
 
+    VkDeviceAddress address() const;
```
Comment (juan-lunarg) [inline tests/vktestbinding.h:535]: Use ` [[nodiscard]] ` attribute.

File: tests/vktestbinding.cpp (hunk 1 above)
Comment (jeremyg-lunarg) [reply tests/vktestbinding.cpp:51]: It could be.

File: tests/vktestbinding.h (hunk 2 above)
Comment (jeremyg-lunarg) [reply tests/vktestbinding.h:535]: I'm not sure about this. vkGetBufferDeviceAddress() doesn't allocate the address, just returns it.  Does it really matter if the caller discards it?

Comment (juan-lunarg) [reply tests/vktestbinding.h:535]: Why would you call the function if you intend to discard the result?

Comment (jeremyg-lunarg) [reply tests/vktestbinding.h:535]: Ok. I don't have a good reason for to not tell people not to do that. :)

Comment (arno-lunarg) [reply tests/vktestbinding.h:535]: Pls do not add `[[nodiscard]]`! For some tests I wrote, I called `vk::GetBufferDeviceAddress` without storing the address, because calling this function will register the buffer and its address in the address -> buffers map.

**Comment by arno-lunarg** [comment]
Nice, I am waiting for this to be merged so that I use those changes in https://github.com/KhronosGroup/Vulkan-ValidationLayers/pull/4861

File: tests/vktestbinding.h (hunk 2 above)
Comment (arno-lunarg) [reply tests/vktestbinding.h:535]: Or maybe it is up to me to add [[maybe_unused]], since what I describe is a corner case...

**Review by Tony-LunarG — APPROVED** [review]
LGTM! Nice Cleanup

File: tests/vklayertests_descriptor_renderpass_framebuffer.cpp [hunk 3]
```diff
@@ -3462,51 +3337,39 @@ TEST_F(VkLayerTest, FramebufferCreateErrors) {
...
-    {
-        if (!multiviewSupported) {
-            printf("VK_KHR_Multiview Extension not supported, skipping tests\n");
-        } else {
-            // Test multiview renderpass with more than 1 layer
-            uint32_t viewMasks[] = {0x3u};
-            auto rpmvci = LvlInitStruct<VkRenderPassMultiviewCreateInfo>(nullptr, 1u, viewMasks, 0u, nullptr, 0u, nullptr);
-            auto rpci_mv = LvlInitStruct<VkRenderPassCreateInfo>(&rpmvci, 0u, 0u, nullptr, 1u, &subpass, 0u, nullptr);
+    if (!multiviewSupported) {
+        printf("VK_KHR_Multiview Extension not supported, skipping tests\n");
```
Comment (arno-lunarg) [inline tests/vklayertests_descriptor_renderpass_framebuffer.cpp:3353]: GTEST_SKIP this?

Comment (jeremyg-lunarg) [reply tests/vklayertests_descriptor_renderpass_framebuffer.cpp:3353]: there's additional checking the test that happens after this statement that should run even without multiview extension

**Review by juan-lunarg — APPROVED** [review]
